SimPEG/EM/Base.py

```diff
@@ -154,6 +197,23 @@ def MeMuI(self):
             self._MeMuI = self.mesh.getEdgeInnerProduct(self.mu, invMat=True)
         return self._MeMuI
 
+    # TODO: This should take a vector
+    def MeMuIDeriv(self, u):
+        """
+        Derivative of :code:`MeSigma` with respect to the model
+        """
+        # TODO: only works for diagonal tensors. getEdgeInnerProductDeriv,
+        #       invMat=True should be implemented in SimPEG
+
+        if (
+            getattr(self, 'muMap', None) is not None or
```

Comment:
Do we need to be careful with the attr not existing?
Do we have to check both mu and mui maps? If one is set the other should be too?

```python
if self.muMap is None:
    return Utils.Zero()
dMeMuI_dI = -self.MeMuI**2
dMe_dmu = self.mesh.getEdgeInnerProductDeriv(self.mu)(u)
return dMeMuI_dI * (dMe_dmu * self.muDeriv)
```

I think that is much cleaner. Would need to validate that `muiMap` not being `None` would still work.